### PR TITLE
DEBUG: Use readdir() instead of readdir_r().

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

File: src/ucs/debug/debug.c [hunk 1]
```diff
@@ -693,13 +693,12 @@ static void ucs_debug_stop_other_threads()
...
     for (;;) {
-        ret = readdir_r(dir, entry, &result);
-        if (ret < 0) {
-            ucs_log_fatal_error("Unable to read from %s: %m", task_dir);
-            break;
-        }
-
-        if (result == NULL) {
+        errno = 0;
+        entry = readdir(dir);
```
Comment (yosefe) [inline src/ucs/debug/debug.c]: need to also remove allocation of entry with alloca and calculation of name_max

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2785/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/208/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2786/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/209/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2181/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2182/ for details (Mellanox internal link).

**Pull request merged**